[PLATFORM-1083] Security Slider

timoxley opened the pull request:
On top of #699 because otherwise my environment doesn't work.

![security-slider](https://user-images.githubusercontent.com/43438/66023102-ba5fcc80-e522-11e9-8ea0-a166629900a0.gif)

juhah commented on app/src/userpages/components/StreamPage/Show/SecurityView/Slider.jsx:
```diff
@@ -0,0 +1,110 @@
+ * Why reading positions from DOM? The center points of the options we're drawing graphics
+ * between is dependent on the width of the label text, which we can't know without rendering.
+ * Without this, the rendered points wouldn't line up with the text or the text wouldn't align with edges of container.
+ * Simple to just less CSS do its thing and piggypack on top of the layout it produces.
```
less = let

juhah commented on app/src/userpages/components/StreamPage/Show/SecurityView/index.jsx:
```diff
@@ -0,0 +1,183 @@
+ * The slider setting maps to these booleans as follows:
+ * Basic:
+ * requireSignedData: false
+ * requireEncryptedData: false
```
Don't know if I have an older e&e instance running but `requireEncryptedData` is not saved or returned but the stream endpoint. If you set the selection to "Encrypted", it does not remember the choice after refreshing the page.

hpihkala commented:
@timoxley #699 merged

timoxley commented:
@juhah 

> Line dots should probably have cursor: pointer since they are clickable
Fixed.

> I don't have access to the design but should the icon in the stream list chance according to selection (signed -> blue icon, encrypted -> green icon)

![image](https://user-images.githubusercontent.com/43438/66104687-f9a62000-e5eb-11e9-8a4f-9515c8357c62.png)

Fixed.

timoxley commented:
> Compared to previous version, icons are not faded anymore when not selected (I assume on purpose).

@juhah was a bug, fixed
